### PR TITLE
add an option in CollateDefault to add fixed items to batch_dict

### PR DESCRIPTION
**Pull request by mosheraboh**

### CODE REVIEW

File: fuse/utils/ndict.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def values(self) -> dict_items:
     def items(self) -> dict_items:
         return self._stored.items()
 
-    def merge(self, other: NDict) -> None:
+    def merge(self, other: MutableMapping) -> None:
```
Comment (SagiPolaczek) [inline fuse/utils/ndict.py:147]: Now that I see it:

Why do we even need `merge()` if we just use the `update()` (which compatible regular dict..)

File: fuse/dl/lightning/pl_module.py [hunk 2]
```diff
@@ -195,13 +195,15 @@ def validation_step(
         else:
             losses = self._losses
         # given the batch_dict and FuseMedML style losses - compute the losses, return the total loss (ignored) and save losses values in batch_dict["losses"]
-        _ = step_losses(losses, batch_dict)
+        if losses is not None:
```
Comment (SagiPolaczek) [inline fuse/dl/lightning/pl_module.py:198]: How `losses` can be `None`? And why the same change shouldn't be in `train` as well?

I thought that we must define one of the both: `self._validation_losses`, `self._losses`. 

Also we have the line:
```python 
        self._losses = losses if losses is not None else {}
```

So maybe we can do the same for `_validation_losses` (or assert that one of them exists?)

Comment (SagiPolaczek) [reply fuse/dl/lightning/pl_module.py:198]: Just throwing thoughts.. I'm not familiar with the motivation :)

**Review by SagiPolaczek — DISMISSED** [review]
Thanks!

I added couple of questions and thoughts inline. 
besides that, LGTM 🚀

File: fuse/utils/ndict.py (hunk 1 above)
Comment (SagiPolaczek) [reply fuse/utils/ndict.py:147]: In the same context, I've send you a cool library on Slack that might be a cool replacement for the loved `NDict`

Comment (mosheraboh) [reply fuse/utils/ndict.py:147]: 🤷 
Lets keep it for backward compatibility though
Saw it,
We started with this library btw, but then we decided to move to NDict cause it was with too many capabilities for our needs.

File: fuse/dl/lightning/pl_module.py (hunk 2 above)
Comment (mosheraboh) [reply fuse/dl/lightning/pl_module.py:198]: loss is a must to train a model (and not for validation).
The code is confusing right now, let me make it clearer. Thanks for the feedback!

File: fuse/utils/ndict.py (hunk 1 above)
Comment (SagiPolaczek) [reply fuse/utils/ndict.py:147]: Cool! good to know! 

Indeed the `NDict` is pretty slim and simple (and thus fast).

**Pull request merged**